Locationapp/app/src/main/java/com/example/manya/locationapp/MainActivity.java

```diff
@@ -1,20 +1,166 @@
 package com.example.manya.locationapp;
 
+import android.content.ContentValues;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.database.Cursor;
+import android.net.Uri;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.ActivityCompat;
 import android.support.v7.app.AppCompatActivity;
 
 import android.os.Bundle;
 import android.support.v7.widget.LinearLayoutManager;
 import android.support.v7.widget.RecyclerView;
-public class MainActivity extends AppCompatActivity {
+import android.util.Log;
+import android.view.View;
+import android.widget.CheckBox;
+import android.widget.Toast;
+
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.GooglePlayServicesNotAvailableException;
+import com.google.android.gms.common.GooglePlayServicesRepairableException;
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.android.gms.common.api.PendingResult;
+import com.google.android.gms.common.api.ResultCallback;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.location.places.Place;
+import com.google.android.gms.location.places.PlaceBuffer;
+import com.google.android.gms.location.places.Places;
+import com.google.android.gms.location.places.ui.PlacePicker;
+import com.example.manya.locationapp.provider.PlaceContract;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static android.webkit.ConsoleMessage.MessageLevel.LOG;
+
+public class MainActivity extends AppCompatActivity implements GoogleApiClient.ConnectionCallbacks,
+        GoogleApiClient.OnConnectionFailedListener {
+    public static final String TAG = MainActivity.class.getSimpleName();
     private PlacesListAdapter Adapter;
```

Comment:
Naming conventions should be followed. IMO

https://github.com/ribot/android-guidelines/blob/master/project_and_code_guidelines.md